web/lib/src/helpers/lists.dart

```diff
@@ -69,3 +70,44 @@ class JSImmutableListWrapper<T extends JSObject, U extends JSObject>
   @override
   U elementAt(int index) => this[index];
 }
+
+/// A wrapper for live node lists. `NodeList` and `HTMLCollection` that are
+/// [live](https://developer.mozilla.org/en-US/docs/Web/API/NodeList#live_vs._static_nodelists)
+/// can be safely modified at runtime. This requires an instance of `P`, a
+/// container that elements would be added to or removed from.
+class JSLiveNodeListWrapper<P extends Node, T extends JSObject, U extends Node>
```

Comment:
`dart:html` has overrides for `first` and `last` for `children` that use `firstElementChild` and `lastElementChild`, respectively.

This I believe is different from `item(0)` and `item(len - 1)`, so I wonder if we should split the implementation here so `childNodes` and `children` work differently.

----

As an aside, `remove` is also overridden in `dart:html` and uses `identical` (which won't work with `dart:js_interop` types and would need `===`/`strictEquals` from `dart:js_interop`). I think the default implementation in `ListMixin` that uses `==` is okay as I don't believe the differences between `==` and `===` can materialize in `children` or `childNodes`.

Reply:
Not sure if this is realiable source:
https://stackoverflow.com/questions/43324751/is-there-a-difference-between-children0-and-firstelementchild
seems difference is only in returned value when list is empty but we throw exceptions in both cases.

I've added `remove`. It was indeed tricky as type of parameter in List interface is `Object?`, please take a look if type checking i used makes sense.